### PR TITLE
feat(duplicate-nodes):  Duplicate steps/container on the canvas

### LINKED ISSUE #2448: Duplicate a Node/Container in the canvas
### Please describe the feature that you want to propose

Duplicating a step node will just duplicate that node as a next node.
Check the scope for duplicating the conatiner.

### PR DESCRIPTION
**Pull request by shivamG640**
Fixes #2448

### CODE REVIEW

File: packages/ui/src/components/Visualization/Custom/ContextMenu/NodeContextMenu.test.tsx [hunk 1]
```diff
@@ -137,6 +144,20 @@ describe('NodeContextMenu', () => {
...
+      isCompatible: true,
+    });
+
+    const wrapper = render(<NodeContextMenu element={element} />, { wrapper: TestWrapper });
+
+    const item = wrapper.getByTestId('context-menu-item-paste-as-next-step');
+
+    expect(item).toBeInTheDocument();
+  });
+
```
Comment (lordrip) [inline packages/ui/src/components/Visualization/Custom/ContextMenu/NodeContextMenu.test.tsx:160]: Can we add a test to check that clicking on the duplicate context item calls duplicate?

File: packages/ui/src/components/Visualization/Canvas/StepToolbar/StepToolbar.tsx [hunk 2]
```diff
@@ -43,9 +45,24 @@ export const StepToolbar: FunctionComponent<IStepToolbar> = ({
...
+          icon={<BlueprintIcon />}
+          className="step-toolbar__button"
+          data-testid="step-toolbar-button-duplicate"
+          variant="control"
+          title="Duplicate"
+          onClick={(event) => {
+            onDuplicate();
+            event.stopPropagation();
+          }}
+        ></Button>
```
Comment (lordrip) [inline packages/ui/src/components/Visualization/Canvas/StepToolbar/StepToolbar.tsx:63]:
```suggestion
        />
```

File: packages/ui/src/models/camel/camel-resource.ts [hunk 3]
```diff
@@ -10,7 +10,8 @@ import { SourceSchemaType } from './source-schema-type';
 export interface CamelResource {
   getVisualEntities(): BaseVisualCamelEntity[];
   getEntities(): BaseCamelEntity[];
-  addNewEntity(entityType?: EntityType): string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  addNewEntity(entityType?: EntityType, entityTemplate?: any): string;
```
Comment (lordrip) [inline packages/ui/src/models/camel/camel-resource.ts:14]: Can we use `unknown` instead?

File: packages/ui/src/models/camel/camel-route-resource.ts [hunk 4]
```diff
@@ -119,11 +123,17 @@ export class CamelRouteResource implements CamelResource, BeansAwareResource {
...
     if (entityType && entityType !== EntityType.Route) {
       const supportedEntity = CamelRouteResource.SUPPORTED_ENTITIES.find(({ type }) => type === entityType);
       if (supportedEntity) {
-        const entity = new supportedEntity.Entity();
+        let entity: BaseVisualCamelEntity;
+        if (entityTemplate) {
+          entity = new supportedEntity.Entity(entityTemplate);
+        } else {
+          entity = new supportedEntity.Entity();
+        }
```
Comment (lordrip) [inline packages/ui/src/models/camel/camel-route-resource.ts:136]: Curious question, shouldn't the following serve the same purpose?

```suggestion
          entity = new supportedEntity.Entity(entityTemplate);
```

**Review by lordrip — COMMENTED** [review]
Would it be possible to avoid showing the duplicate button for `Kamelets` and `Pipes`?

image

**Comment by lordrip** [comment]
@shivamG640 I need to merge this PR so I can move forward with the XML sorting one. Since most of the comments are small things, we can tackle them in a separate PR.
